.github/workflows/deploy.yml

```diff
@@ -0,0 +1,67 @@
+name: deploy
+
+
+# Only trigger, when the test workflow succeeded
+"on":
+  push:
+    branches: [main]
+  workflow_run:
+    workflows: ["test"]
+    types:
+      - completed
+
+
+jobs:
+  deploy:
+    # this job will only run if the PR has been merged
+    if: github.event.pull_request.merged == true
```

Comment:
Hmm, is this necessary, did you check? The "main" branch is where PRs land in, not e.g. where they're created from. Plus we also want to run the CI and the deployment if commits where pushed to "main" directly, without a PR.